### PR TITLE
Do not wait for messaging unless the document is embedded

### PR DESCRIPTION
**Pull request by dvoytenko**

### CODE REVIEW

**Comment by erwinmombay** [comment]
@dvoytenko LGTM

File: src/service/viewer-impl.js [hunk 1]
```diff
@@ -253,48 +253,40 @@ export class Viewer {
...
-
     /**
      * This promise will resolve when communications channel has been
-     * established or timeout in 5 seconds. The timeout is needed to avoid
+     * established or timeout in 20 seconds. The timeout is needed to avoid
      * this promise becoming a memory leak with accumulating undelivered
-     * messages.
-     * @private @const {!Promise<!Viewer>}
+     * messages. The promise is only available when the document is embedded.
+     * @private @const {?Promise}
```
Comment (erwinmombay) [inline src/service/viewer-impl.js]: remove const

File: src/service/viewer-impl.js [hunk 2]
```diff
@@ -253,48 +253,40 @@ export class Viewer {
...
+    }
 
     /**
      * A promise for non-essential messages. These messages should not fail
      * if there's no messaging channel set up. But ideally viewer would try to
-     * deliver if at all possible.
-     * @private @const {!Promise<!Viewer>}
+     * deliver if at all possible. This promise is only available when the
+     * document is embedded.
+     * @private @const {?Promise}
```
Comment (erwinmombay) [inline src/service/viewer-impl.js]: ditto, remove const.

**Comment by erwinmombay** [comment]
@dvoytenko minor ask for `@const` removal.

**Comment by erwinmombay** [comment]
or we could inline it so we can keep `@const` :dash:

File: src/service/viewer-impl.js (hunk 1 above)
Comment (dvoytenko) [inline src/service/viewer-impl.js]: Good idea. Inlined.

File: src/service/viewer-impl.js (hunk 2 above)
Comment (dvoytenko) [inline src/service/viewer-impl.js]: Inlined

File: src/service/viewer-impl.js [hunk 3]
```diff
@@ -253,48 +253,36 @@ export class Viewer {
...
-        });
+    this.messagingReadyPromise_ = this.isEmbedded_ ?;
+        timer.timeoutPromise(
+            20000,
+            new Promise(resolve => {
+              /** @private @const {function()|undefined} */
+              this.messagingReadyResolver_ = resolve;
+            })).catch(reason => {
+              throw getChannelError(reason);
+            }) : null;
```
Comment (jridgewell) [inline src/service/viewer-impl.js]: Would it be better to return a rejected promise? That'll avoid all the conditionals below.

Comment (dvoytenko) [inline src/service/viewer-impl.js]: Well, that's how it was before ;) Somewhere these conditionals creep in. I guess the main goal is to defer any rejection until the API is actually improperly requested.

**Pull request merged**